content/en/examples/controllers/daemonset.yaml

```diff
@@ -15,6 +15,8 @@ spec:
         name: fluentd-elasticsearch
     spec:
       tolerations:
+      # this toleration is to have the daemonset runnable on master nodes
+      # remove it if your masters can't run pods
```

Comment:
@sam-sla 

I couldn't find an authoritative page on this with a quick Google, but I have a feeling that `node-role.kubernetes.io/master` is _not_ a [well-known label](https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/) and that it's more of a convention. `kubeadm` certainly adds that label and many similar tools do as well.

Not all clusters have master nodes (you don't have to run the control plane components inside Pods, even). So this is a bit of a subtle thing to explain well.

The good news is that adding the toleration isn't likely to make things any worse if your control plane isn't running on master nodes, or if your control plane nodes have a different label.

Does that make sense? I'm happy to go into more detail.

Reply:
Sorry I am not an expert, let me try to explain my issue:
Kubeadm seems to add that label and taint by default https://kubernetes.io/docs/reference/setup-tools/kubeadm/implementation-details/#mark-master 
In my case I had a cluster with a master node (and therefore that label and taint which I wasn't aware of) and two other nodes. Our deployment has a recently added daemonset, which used this daemonset.yaml example as its base, and that scheduled 3 pods, one of them on the master node (but failed because it couldn't pull the image due to secrets I think).
Indeed on clusters without master nodes this toleration has no effect, but on a cluster with master nodes this default will schedule pods there and that might fail.
Do you think this can be a broad issue for the users, or we have something exotic with our configuration?

Reply:
`kubeadm` is very common as a tool to manage clusters.

I think it's worth highlighting to readers that they might or might not want that DaemonSet to schedule onto their control plane hosts, and hint at how to control that.
I think it's quite likely that readers _would_ expect a DaemonSet to schedule onto control plane nodes.

Reply:
core kubernetes does not know about the `node-role.kubernetes.io/master` taint.
some might even want to remove the `node-role.kubernetes.io/master` example from this YAML as a whole.

> remove it if your masters can't run pods

even if the taint key includes the word "master". usage of the word in documentation is discouraged.
instead use "control-plane Nodes".

i'm leaning towards not adding this comment here, and deferring to SIG Apps whether they want to clean the `node-role.kubernetes.io/master` examples at some point in time.

Reply:
My first intention was to actually remove the toleration example but then I found that other example with this kind of comment and I went for the less destructive approach. Fell free to forward the issue and even close the PR if not applicable, or let me know what I can do :)